lib/arproxy/query_context.rb

```diff
@@ -0,0 +1,18 @@
+module Arproxy
+  class QueryContext
+    attr_accessor :raw_connection, :execute_method_name, :with_binds, :name, :binds, :kwargs
+
+    def initialize(raw_connection:, execute_method_name:, with_binds:, name: nil, binds: [], kwargs: {})
+      @raw_connection = raw_connection
+      @execute_method_name = execute_method_name
+      @with_binds = with_binds
+      @name = name
+      @binds = binds
+      @kwargs = kwargs
+    end
+
+    def with_binds?
+      !!@with_binds
```

Comment:
imo: How about defining `#with_binds?` as `!!@binds` and set `binds: nil` in the initializer for non-bind executions? Currently `QueryContext#binds` is meaningless when `!with_binds?` and it is less intuitive for users I think

Reply:
I also considered the same thing, but I wonder if some adapter might use `binds: nil` for something at some point.

Reply:
hmm right

Reply:
@nekketsuuu is correct. `@with_binds` is used to identify whether the target method has `binds` as an argument. This value needs to be managed separately from whether it is `nil` or not.

e.g. An execute method without binds:

https://github.com/rails/rails/blob/v7.1.0/activerecord/lib/active_record/connection_adapters/mysql2/database_statements.rb#L96

with binds:

https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L552